src/test/java/com/google/firebase/database/FirebaseDatabaseTest.java

```diff
@@ -171,4 +174,135 @@ public void testInitAfterAppDelete() throws ExecutionException, InterruptedExcep
     assertNotNull(db2);
     assertNotSame(db1, db2);
   }
+
+  @Test
+  public void testDbUrlIsEmulatorUrlWhenSettingOptionsManually() {
+    class CustomTestCase {
+
+      private String suppliedDbUrl;
+      private String envVariableUrl;
+      private String expectedEmulatorUrl;
+      private String namespace;
+
+      private CustomTestCase(String suppliedDbUrl, String envVariableUrl,
+          String expectedEmulatorUrl, String namespace) {
+        this.suppliedDbUrl = suppliedDbUrl;
+        this.envVariableUrl = envVariableUrl;
+        this.expectedEmulatorUrl = expectedEmulatorUrl;
+        this.namespace = namespace;
+      }
+    }
+
+    List<CustomTestCase> testCases;
+    testCases = ImmutableList.of(
+        // cases where the env var is ignored because the supplied DB URL is a valid emulator URL
+        new CustomTestCase("http://my-custom-hosted-emulator.com:80?ns=dummy-ns", "",
+            "http://my-custom-hosted-emulator.com:80", "dummy-ns"),
+        new CustomTestCase("http://localhost:9000?ns=test-ns", null,
+            "http://localhost:9000", "test-ns"),
+
+        // cases where the supplied DB URL is not an emulator URL, so we extract ns from it
+        // and append it to the emulator URL from env var(if it is valid)
+        new CustomTestCase("https://valid-namespace.firebaseio.com", "localhost:8080",
+            "http://localhost:8080", "valid-namespace"),
+        new CustomTestCase("https://test.firebaseio.com?ns=valid-namespace", "localhost:90",
+            "http://localhost:90", "valid-namespace")
+    );
+    boolean earlierDefaultAppFound = false;
+    try {
+      FirebaseApp.getInstance(FirebaseApp.DEFAULT_APP_NAME).delete();
+      earlierDefaultAppFound = true;
```

Comment:
We might be able to remove this if you remove the `@Before` method and inline it to the tests that need it.

Reply:
Done